components/object_store/src/disk_cache.rs

```diff
@@ -296,40 +453,72 @@ impl DiskCacheStore {
         underlying_store: Arc<dyn ObjectStore>,
         partition_bits: usize,
     ) -> Result<Self> {
-        let page_num_per_part = cap / page_size;
-        ensure!(page_num_per_part != 0, InvalidCapacity);
-        let _ = Self::create_manifest_if_not_exists(&cache_dir, page_size).await?;
-        let cache = DiskCache::try_new(cache_dir.clone(), page_num_per_part, partition_bits)?;
+        let page_num = cap / page_size;
+        ensure!(page_num != 0, InvalidCapacity);
+
+        let manifest = Self::create_manifest_if_not_exists(&cache_dir, page_size).await?;
+        if !manifest.is_valid(CURRENT_VERSION, page_size) {
+            Self::reset_cache(&cache_dir, page_size).await?;
+        }
+
+        let cache = DiskCache::try_new(cache_dir.clone(), page_num, partition_bits)?;
         Self::recover_cache(&cache_dir, &cache).await?;
 
-        let size_cache = Arc::new(Mutex::new(LruCache::new(cap / page_size)));
+        let init_size_lru = |partition_num| -> Result<_> {
+            let cap_per_part = FILE_SIZE_CACHE_CAP / partition_num;
+            assert!(cap_per_part > 0);
+            Ok(LruCache::new(cap_per_part))
+        };
```

Comment:
For more precise memory control, need div FILE_MATA_DATA_SIZE. `FILE_SIZE_CACHE_CAP / partition_num / FILE_MATA_DATA_SIZE`

Reply:
The `FILE_SIZE_CACHE_CAP` here is the number of the cache entry rather than the size. Besides, the size of an entry in this cache is dynamic but small enough, that is to say, there is no need and no way to impose control over the memory usage by that.